zebra/if_netlink.c

```diff
@@ -350,11 +350,38 @@ static void netlink_vrf_change(struct nlmsghdr *h, struct rtattr *tb,
 			}
 		}
 
-		/*
-		 * vrf_get is implied creation if it does not exist
+		/*Treat VRF add for existing vrf as update
+		 * Update VRF ID and also update in VRF ID table
 		 */
-		vrf = vrf_get((vrf_id_t)ifi->ifi_index,
-			      name); // It would create vrf
+		if (name)
+			vrf = vrf_lookup_by_name(name);
+		if (vrf && (vrf_id_t)ifi->ifi_index != VRF_UNKNOWN
+		    && vrf->vrf_id != VRF_UNKNOWN
+		    && vrf->vrf_id != (vrf_id_t)ifi->ifi_index) {
+			zlog_info(
```

Comment:
This should be a guarded debug statement.

Reply:
I have done the changes.